internal/internal_event_handlers.go

```diff
@@ -1067,38 +1077,48 @@ func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 	eventID int64,
 	attributes *eventpb.MarkerRecordedEventAttributes,
 ) error {
-	encodedValues := newEncodedValues(attributes.Details, weh.dataConverter)
 	switch attributes.GetMarkerName() {
 	case sideEffectMarkerName:
-		var sideEffectID int64
-		var result *commonpb.Payloads
-		_ = encodedValues.Get(&sideEffectID, result)
-		weh.sideEffectResult[sideEffectID] = result
+		for sideEffectID, result := range attributes.GetDetails() {
+			weh.sideEffectResult[sideEffectID] = result
+		}
 		return nil
 	case versionMarkerName:
-		var changeID string
-		var version Version
-		_ = encodedValues.Get(&changeID, &version)
-		weh.changeVersions[changeID] = version
-		weh.decisionsHelper.handleVersionMarker(eventID, changeID)
+		for changeID, versionPayload := range attributes.GetDetails() {
+			var version Version
+			_ = weh.dataConverter.FromData(versionPayload, &version)
+			weh.changeVersions[changeID] = version
+			weh.decisionsHelper.handleVersionMarker(eventID, changeID)
+		}
 		return nil
 	case localActivityMarkerName:
-		return weh.handleLocalActivityMarker(attributes.Details)
+		err := weh.handleLocalActivityMarker(attributes.GetDetails(), attributes.GetFailure())
+		if err != nil {
+			return fmt.Errorf("marker name %q: %w", localActivityMarkerName, err)
+		}
+		return nil
 	case mutableSideEffectMarkerName:
-		var fixedID string
-		var result *commonpb.Payloads
-		_ = encodedValues.Get(&fixedID, result)
-		weh.mutableSideEffect[fixedID] = result
+		for fixedID, result := range attributes.GetDetails() {
+			weh.mutableSideEffect[fixedID] = result
+		}
 		return nil
 	default:
-		return fmt.Errorf("unknown marker name \"%v\" for eventId \"%v\"",
-			attributes.GetMarkerName(), eventID)
+		return fmt.Errorf("marker name %q for eventId %d: %w", attributes.GetMarkerName(), eventID, ErrUnknownMarkerName)
 	}
 }
 
-func (weh *workflowExecutionEventHandlerImpl) handleLocalActivityMarker(markerData *commonpb.Payloads) error {
-	lamd := localActivityMarkerData{}
+func (weh *workflowExecutionEventHandlerImpl) handleLocalActivityMarker(details map[string]*commonpb.Payloads, failure *failurepb.Failure) error {
+	if details == nil {
+		return ErrMissingLocalActivityMarkerDetails
+	}
 
+	var markerData *commonpb.Payloads
+	var ok bool
+	if markerData, ok = details[localActivityMarkerDataDetailsName]; !ok {
```

Comment:
What about failure? Are details always popluated in this case?

Reply:
Yes, `result` will not be there, but `data` is always added.